### PR TITLE
Script for updating configuration file from v1 -> v2

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: scripts/migrations/configuration_v1_to_v2.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    # Apply updates
+    del config['domains']  # deprecated
+    config['domain'] = domian
+    config['version'] = 2
+
+    # Commit updates
+    with open(filepath, 'w') as file:
+        file.write(json.dumps(config, indent=4))
+    print('OK! Migrated configuration file from v1 -> v2.')
```
Comment (derekpierre) [inline scripts/migrations/configuration_v1_to_v2.py:59]: 💥

**Review by derekpierre — APPROVED** [review]
🎸 - (haven't ran it though)

File: scripts/migrations/configuration_v1_to_v2.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    """Updates configuration file V2 to V2 by remapping 'domains' to 'domain'"""
+
+    # Read + deserialize
+    with open(filepath, 'r') as file:
+        contents = file.read()
+    config = json.loads(contents)
+
+    # Check we have version 1 indeed
+    if config['version'] != 1:
+        raise RuntimeError('Existing configuration is not version 1.')
```
Comment (derekpierre) [inline scripts/migrations/configuration_v1_to_v2.py:38]:
```suggestion
        raise RuntimeError(f'Existing configuration is not version 1; instead was {config['version']}.')
```

**Review by derekpierre — APPROVED** [review]
Ran a test. The only minor thing was version was no longer last, but fairly irrelevant:

i.e.
My old file was :
```
{
  "federated_only": null,
  "checksum_address": null,
  "keyring_root": "/Users/derek/Library/Application Support/nucypher/keyring",
  "domains": [
    "gemini"
  ],
  ...
  "availability_check": false,
  "version": 1
}
```

and my new file became
```
{
    ...
    "availability_check": false,
    "version": 2,
    "domain": "gemini"
}
```

File: scripts/migrations/configuration_v1_to_v2.py (hunk 2 above)
Comment (KPrasch) [reply scripts/migrations/configuration_v1_to_v2.py:38]: :heavy_check_mark:

**Review by cygnusv — APPROVED** [review]
First of a kind!

**Review by KPrasch — COMMENTED** [review]
Plenty of room for improvement - but a step in the right direction no less.

**Pull request merged**